R/estimateAlpha.R

```diff
@@ -0,0 +1,245 @@
+    }
+    # Looping over the vector of indices to be estimated
+    for (i in seq_along(index)) {
+        # Getting the corresponding alpha indices function by parsing the index
+        FUN <- NULL
+        if(any(grepl(index[i], .get_indices("diversity")))) {
+            # making name having the alpha type suffix or leave it as is if 
+            # user defined
+            name[i] <- .parse_name(index[i], name[i], "diversity")
+            # cleaning index from suffix to be used with the corresponding index
+            # function
+            index[i] <- gsub("_diversity", "", index[i])
+            FUN <- .estimate_diversity
+        } else if(any(grepl(index[i], .get_indices("dominance")))) {
+            name[i] <-  .parse_name(index[i], name[i], "dominance")
+            index[i] <- gsub("_dominance", "", index[i])
+            FUN <- .estimate_dominance
+        } else if (any(grepl(index[i], .get_indices("evenness")))) {
+            name[i] <- .parse_name(index[i], name[i], "evenness")
+            if (index[i]!="simpson_evenness") {
+                index[i] <- gsub("_evenness", "", index[i])
+            }
+            FUN <- .estimate_evenness
+        } else if (any(grepl(index[i], .get_indices("richness")))) {
+            name[i] <- .parse_name(index[i], name[i], "richness")
+            index[i] <- gsub("_richness", "", index[i])
+            FUN <- .estimate_richness
+        } else {
+            stop("'index' is coresponding to none of the alpha diversity indices.
+                  'index' should be one of: coverage_diversity, fisher_diversity,
+                  faith_diversity, gini_simpson_diversity,
+                  inverse_simpson_diversity, log_modulo_skewness_diversity,
+                  shannon_diversity, absolute_dominance, dbp_dominance,
+                  core_abundance_dominance, gini_dominance,
+                  dmn_dominance, relative_dominance, simpson_lambda_dominance,
+                  camargo_evenness, pielou_evenness, simpson_evenness,
+                  evar_evenness, bulla_evenness, ace_richness, chao1_richness,
+                  hill_richness or observed_richness.",
+                 call. = FALSE)
+        }
+        # Performing rarefaction if rarefaction.depth is specified to be less 
+        # the max of the total read counts
+        if (rarefaction.depth < max(colSums(assay(x, assay.type)), na.rm = TRUE)) {
+            x <- .alpha_rarefaction(x, n.iter = n.iter,
+                                    args.sub = list(assay.type=assay.type,
+                                                    min_size=rarefaction.depth,
+                                                    verbose=FALSE),
+                                    FUN=FUN,
+                                    args.fun=list(index=index[i], assay.type="subsampled"),
+                                    ...,
+                                    name=name[i])
+        } else {
+            # Estimate index without rarefaction
```

Comment:
Why there is a warning? Add comment

Reply:
suppressing warnings should be avoided. Probably there is a good reason for that, but it might be good to add info why it is done